<?xml version="1.0" ?>

<launch>
    <arg name="robot_type" default="$(env ROBOT_TYPE)" doc="Robot type: [lt, go1, ti5]"/>
    <arg name="rviz" default="false"/>
    <arg name="urdfFile"          default="$(find legged_ti5_description)/urdf/ti5_column.urdf"/>
    <arg name="description_name"   default="legged_robot_description"/>
    <arg name="record_data" default="false"/>

    <param name="test_amplitude1" value="20.0" />
    <param name="test_amplitude2" value="20.0" />
    <param name="test_motor_id" value="2" />
    
    <param name="use_imu_topic" value="false" />
    <param name="robot_description" textfile="$(find legged_ti5_description)/urdf/ti5.urdf" />
    <include file="$(find legged_gazebo)/launch/$(env ROBOT_TYPE)_empty_world.launch"/>

    <group if="$(arg rviz)">
        <param name="$(arg description_name)" textfile="$(arg urdfFile)"/>
        <arg name="rvizconfig" default="$(find rl_controllers)/rviz/legged_robot.rviz" />
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rvizconfig)" output="screen" />
      </group>

      <group if="$(arg record_data)">
        <include file="$(find rl_controllers)/launch/record_data.launch"/>
    </group>

    <include file="$(find rl_controllers)/launch/load_ac_controller.launch"/>

</launch>